### PR TITLE
Add description_hash in createinvoice

### PR DESCRIPTION
**Pull request by btcontract**
Fixes https://github.com/ACINQ/eclair/issues/1816.

### CODE REVIEW

**Comment by btcontract** [comment]
Failed `eclair-front` tests pass locally so I'm a bit lost on what's going on, help would be appreciated.

**Comment by t-bast** [comment]
> Failed eclair-front tests pass locally so I'm a bit lost on what's going on, help would be appreciated.

We noticed that we have a few flaky tests in the eclair-front test suite, I restarted the build.
We'll work on fixing the build flakiness.

File: eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala [hunk 1]
```diff
@@ -271,16 +271,13 @@ class PaymentRequestSerializer extends CustomSerializerOnly[PaymentRequest](_ =>
...
     )
     val fieldList = List(JField("prefix", JString(p.prefix)),
       JField("timestamp", JLong(p.timestamp)),
       JField("nodeId", JString(p.nodeId.toString())),
       JField("serialized", JString(PaymentRequest.write(p))),
-      JField("description", JString(p.description match {
-        case Left(l) => l
-        case Right(r) => r.toString()
-      })),
+      p.description.fold(string => JField("description", JString(string)), hash => JField("descriptionHash", JString(hash.toHex))),
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala:280]: 👍

**Review by pm47 — APPROVED** [review]
LGTM. Nice usage of `fold`!

**Pull request merged**